packages/ui/origin-ui-core/src/components/admin/AdminClaimsTable.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    fetchAllDevices,
+    getAllDevices,
+    getBackendClient,
+    getCertificatesClient,
+    getEnvironment
+} from '../../features';
+import { TableMaterial } from '../Table';
+
+interface IRow {
+    id: number;
+    device: string;
+    energy: string;
+    beneficiary: string;
+}
+
+export function AdminClaimsTable(): JSX.Element {
+    const [rows, setRows] = useState([]);
+    const certificatesClient = useSelector(getCertificatesClient);
+    const backendClient = useSelector(getBackendClient);
+    const deviceClient = backendClient?.deviceClient;
+    const dispatch = useDispatch();
+    const environment = useSelector(getEnvironment);
+
+    const allDevices = useSelector(getAllDevices);
+
+    async function update() {
+        const { data: certs } = await certificatesClient.getAll();
+
+        if (certs) {
+            setRows(
+                certs.map((c) => {
+                    const certDevice = allDevices.find((device) =>
+                        device.externalDeviceIds.some(
+                            (deviceExternalId) =>
```

Comment:
Please, avoid doing such long calculation inside `setState`. 
Shape the data first, assign it to a variable and pass as argument to setState function.